cyvcf2/cyvcf2.pyx

```diff
@@ -410,7 +411,7 @@ cdef class VCF(HTSFile):
                     bcf_destroy(b)
                     break
                 if bcf_subset_format(self.hdr, b) != 0:
-                    sys.stderr.write("could not subset variant")
+                    warnings.warn("could not subset variant")
```

Comment:
This needs a test case.

Reply:
I checked the source of `bcf_subset_format()` and found it always succeeds (returning 0). So I removed the code path and put in an assertion instead.